symbolic-base/src/ZkFold/Symbolic/Data/Hash.hs

```diff
@@ -38,6 +40,8 @@ data Hash h a = Hash
 
 instance (SymbolicOutput h, SymbolicOutput a) => SymbolicData (Hash h a)
 instance (SymbolicInput h, SymbolicInput a) => SymbolicInput (Hash h a)
+instance (c ~ (Context h), Conditional (Bool c) h, Symbolic c, SymbolicData a) => Conditional (Bool c) (Hash h a)
+instance (c ~ (Context h), Symbolic c, SymbolicData a, BooleanOf h ~ Bool c, Eq h) => Eq (Hash h a)
```

Comment:
The `Symbolic c` constraint can be inferred from `c ~ Context h`, so it can be removed
Also `c` can be substituted into `Bool c`, removing its mention completely:
```haskell
(SymbolicData a, Eq h, BooleanOf h ~ Bool (Context h))
```